src/test/java/integration/IntegrationTest.java

```diff
@@ -3,563 +3,1428 @@
-        testFilePull(); // Should have no changes
-        testFileChange();
-        testFilePull(); // Should have 1 change
-        testFileMove();
-        testFileChange();
-        testFileRename();
-        testFileChange();
-        testFilePull(); // Should have 3 changes
-        testProjectGetFiles(); // should have 1 file
-
-        testOtherUserRegisterAndLogin();
-        testInvalidAccess();
-        testProjectGrantReadPermission();
-        testInvalidProjectWrite();
-        testProjectUpdatePermission();
-        testValidProjectWrite();
-        testValidRevokePermission();
-        testInvalidAccess();
-
-        testProjectSubscribe();
-        testFileDelete();
-        testProjectGetFiles(); // should have 0 files
-        testProjectUnsubscribe();
-        testProjectSubscribe();
-        testProjectLookup();
-        testUserProjects();
-        testProjectDelete();
-        testUserProjects();
-        // ProjectGetOnlineClientsRequest (NOT IMPLEMENTED)
+            waiter.release(1);
+        }, errHandler);
 
-        testUserDelete();
+        ws2.sendRequest(req);
+        if (!waiter.tryAcquire(1, 5, TimeUnit.SECONDS)) {
+            Assert.fail("Acquire timed out");
+        }
 
-        // Run invalid method type, expect error
-        // Re-run login, expect error(?)
-        // Run invalid login, expect error
+        req = new FileCreateRequest("dummyFileName", "dummyFilePath", proj2.getProjectID(), "dummyData".getBytes()).getRequest(response -> {
+            Assert.assertNotEquals("Failed to throw permissions error on lack of write permissions", 200, response.getStatus());
 
-        Thread.sleep(1000);
-    }
+            waiter.release(1);
+        }, errHandler);
 
-    private void registerFileChangeNotificationHandler(String[] changes, Semaphore waiter) {
-        wsMgr.registerNotificationHandler("File", "Change", notification -> { // Create notification handler
-            Assert.assertEquals("FileChangeNotification gave wrong file ID", fileID, notification.getResourceID());
-            Assert.assertArrayEquals("FileChangeNotification gave wrong changes", changes, ((FileChangeNotification) notification.getData()).changes);
-            Assert.assertEquals("FileChangeNotification gave wrong file version", fileVersion + 1, ((FileChangeNotification) notification.getData()).fileVersion);
-            Assert.assertEquals("FileChangeNotification gave wrong base file version", fileVersion, ((FileChangeNotification) notification.getData()).baseFileVersion);
+        ws1.sendRequest(req);
+        if (!waiter.tryAcquire(1, 5, TimeUnit.SECONDS)) {
+            Assert.fail("Acquire timed out");
+        }
 
-            wsMgr.deregisterNotificationHandler("File", "Change");
-            waiter.release();
-        });
+        /*
+         * Check duplicate file
+         */
+        // TODO: Apparently the server doesn't check duplicate files; this does not work...
```

Comment:
We should track this on Trello to make sure that we don't forget to add it back in in a different PR once it's fixed on the server.